nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/webapp/js/nf/canvas/controllers/nf-ng-canvas-flow-status-controller.js

```diff
@@ -400,6 +406,554 @@
+                            }
+                        }
+
+                        function openRuleMoreInfoDialog() {
+                            var rule = response.rules.find(function(rule){ 
+                                return rule.id === violationInfo.ruleId;
+                            })
+                            $('#violation-menu').hide();
+                            $('#violation-type-pill').empty()
+                                                    .removeClass()
+                                                    .addClass(violationInfo.enforcementPolicy.toLowerCase() + ' violation-type-pill')
+                                                    .append(violationInfo.enforcementPolicy);
+                            $('#violation-display-name').empty().append(violationInfo.violationMessage);
+                            $('#violation-description').empty().append(rule.descriptors['component-type'].description);
+                            $('#violation-menu-more-info-dialog').modal( "show" );
+                            $('.violation-docs-link').click(function () {
+                                // open the documentation for this flow analysis rule
+                                nfShell.showPage('../nifi-docs/documentation?' + $.param({
+                                    select: rule.type,
+                                    group: rule.bundle.group,
+                                    artifact: rule.bundle.artifact,
+                                    version: rule.bundle.version
+                                })).done(function () {});
+                            });
+                            unbindViolationMenuHandling();
+                        }
+
+                        function goToComponent() {
+                            $('#violation-menu').hide();
+                            nfCanvasUtils.showComponent(groupId, violationInfo.subjectId);
+                            unbindViolationMenuHandling();
+                        }
+
+                        function unbindViolationMenuHandling() {
+                            $('#violation-menu-go-to').unbind('click', goToComponent);
+                            $('#violation-menu-more-info').unbind('click', openRuleMoreInfoDialog);
+                            $(document).unbind('click', closeViolationWindow);
+                        }
+                    });
+                },
+
+                /**
+                 * Initialize the flow analysis controller.
+                 */
+                init: function () {
+                    var flowAnalysisCtrl = this;
+                    var drawer = $('#flow-analysis-drawer');
+                    var requiredRulesEl = $('#required-rules');
+                    var recommendedRulesEl = $('#recommended-rules');
+                    var newFlowAnalsysisBtnEl = $('#flow-analysis-check-now-btn');
+                    var flowAnalysisRefreshIntervalSeconds = 5;
```

Comment:
Is there a constant elsewhere used for polling that I should be using here?  `5` is an arbitrary value used for development. I'm guessing we don't want polling that often.

Reply:
In `nf-canvas-bootstrap.js` there is a variable `autoRefreshIntervalSeconds` that you'll have to plumb down a few layers starting from [1]. Try setting a breakpoint in the `init` function in `nf-ng-canvas-flow-status-controller.js` and you see how it needs to be plumbed down.

[1] https://github.com/apache/nifi/blob/main/nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/webapp/js/nf/canvas/nf-canvas-bootstrap.js#L353

Reply:
@mcgilman Based on your comment above, I ended up creating a getter/setter in nf-common and calling the set function in nf-canvas-bootstrap. If the approach you had in mind is much different, let me know and I can update accordingly.

Reply:
The change is fine. But I believe it could have been passed down through the `init` functions.